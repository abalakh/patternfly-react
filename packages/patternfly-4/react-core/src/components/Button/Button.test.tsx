import { Button, ButtonVariant } from './Button';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { CartArrowDownIcon } from '@patternfly/react-icons';

Object.values(ButtonVariant).forEach(variant => {
  test(`${variant} button`, () => {
    const view = shallow(
      <Button variant={variant} aria-label={variant}>
        {variant} Button
      </Button>
    );
    expect(view).toMatchSnapshot();
  });
});

test('it adds an aria-label to plain buttons', () => {
  const label = 'aria-label test';
  const view = mount(<Button aria-label={label} />);
  expect(view.find('button').props()['aria-label']).toBe(label);
});

test('link with icon', () => {
  const view = shallow(
    <Button variant={ButtonVariant.link} icon={<CartArrowDownIcon />}>
      Block Button
    </Button>
  );
  expect(view).toMatchSnapshot();
});

test('isBlock', () => {
  const view = shallow(<Button isBlock>Block Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isDisabled', () => {
  const view = shallow(<Button isDisabled>Disabled Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isFocus', () => {
  const view = shallow(<Button isFocus>Focused Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isHover', () => {
  const view = shallow(<Button isHover>Hovered Button</Button>);
  expect(view).toMatchSnapshot();
});

test('isInline', () => {
  const view = shallow(
    <Button variant={ButtonVariant.link} isInline>
      Hovered Button
    </Button>
  );
  expect(view).toMatchSnapshot();
});

test('allows passing in a string as the component', () => {
  const component = 'a';
  const view = mount(<Button component={component} />);
  expect(view.find(component).type()).toBe(component);
});

test('allows passing in a React Component as the component', () => {
  const Component = () => <div>im a div</div>;
  const view = mount(<Button component={Component} />);
  expect(view.find(Component).type()).toBe(Component);
});

test('aria-disabled is set to true and tabIndex to -1 if component is not a button and is disabled', () => {
  const view = shallow(
    <Button component="a" isDisabled>
      Disabled Anchor Button
    </Button>
  );
  expect(view).toMatchSnapshot();
});
