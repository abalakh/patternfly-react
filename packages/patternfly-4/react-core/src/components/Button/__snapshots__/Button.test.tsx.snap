// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aria-disabled is set to true and tabIndex to -1 if component is not a button and is disabled 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`danger button 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`isBlock 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`isDisabled 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`isFocus 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`isHover 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`isInline 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`link button 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`link with icon 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`plain button 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`primary button 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`secondary button 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`tertiary button 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
